reconcile/ocm/types.py

```diff
@@ -0,0 +1,54 @@
+from pydantic import BaseModel, Field, Extra
+from typing import Optional, Union
+
+
+class OCMClusterAutoscale(BaseModel):
+    min_replicas: int
+    max_replicas: int
+
+
+class OCMClusterNetwork(BaseModel):
+    type: Optional[str]
+    vpc: str
+    service: str
+    pod: str
+
+
+class OCMClusterSpec(BaseModel):
+    autoscale: Optional[OCMClusterAutoscale]
+    channel: str
+    disable_user_workload_monitoring: Optional[bool]
+    external_id: Optional[str]
+    id: Optional[str]
+    instance_type: str
+    multi_az: bool
+    nodes: Optional[int]
+    private: bool
+    product: str
+    provider: str
+    provision_shard_id: Optional[str]
+    region: str
+    initial_version: Optional[str]
+    version: str
+
+
+class OSDClusterSpec(OCMClusterSpec, extra=Extra.forbid):
+    load_balancers: int
+    storage: int
+
+
+class ROSAClusterSpec(OCMClusterSpec, extra=Extra.forbid):
```

Comment:
nit: Sorry for nagging about this again :sweat_smile: Would it make sense to have this parameter set in the `OCMSpec.Config` instead? I feel like having it close to `smart_union = True` makes it easier to understand what this is needed for. Further it wouldn't need to be added to class in the `Union`.

```
class OCMSpec(BaseModel):
    
    ...
    
    class Config:
        smart_union = True
        extra = Extra.forbid
```

Or is there a reason to not forbid extra fields for the whole `OCMSpec` class?

Reply:
I followed the example on the docs but I agree with you. I didn't see this option was available in the Config class ...

I can not set it in the OCMSpec class because the dictionary filled with the graphql query has more fields than the class and throws a validation error. I've set it in the OCMClusterSpec which is the base class for the classes in the Union.